shared/general/container-ssh.md

```diff
@@ -43,35 +41,50 @@ $ balena ssh 192.168.1.23
-
-## Add an SSH key to {{ $names.cloud.lower }}
-
-To add an SSH key, go to the _Preferences_ page of {{ $names.cloud.lower }} and select the _SSH Keys_ tab.
-
-![SSH key preferences](/img/common/main_dashboard/eekVBTI.png)
-
-You may either import an existing SSH key from GitHub or manually enter the public SSH key of an existing SSH key on your development machine.
-
-If you do not have an existing key, you can follow [GitHub's documentation][github-ssh], skipping the step about adding the key to your GitHub account, and instead adding the key to your {{ $names.cloud.lower }} account.
+This should work without further configuration in the case of devices running a [development
+{{ $names.os.lower }} image][development-image], which allows unauthenticated `root` access
+(and for this reason, should never be directly exposed to the public internet). In the case of
+a production {{ $names.os.lower }} image, an SSH key must have been previously added to the
+device's `config.json` file, [sshKeys section][config-json-ssh].
 
 ## Using a standalone SSH client
 
-If you prefer to use a standalone SSH client to connect to the device, the SSH server on a device listens on TCP port `22222`. While development images have passwordless root access enabled, production images require an SSH key to be added to the `config.json` file.
+The SSH server of a {{ $names.os.lower }} device (host OS) listens on TCP port `22222`.
+Passwordless access is also possible with a standalone ssh client:
 
 ```shell
-$ ssh -p 22222 root@<device_ip_address>
+$ ssh -p 22222 <username>@<device_ip_address>
 ```
 
+When the username is `root`, [production {{ $names.os.lower }} images][development-image]
+perform authentication against public SSH keys previously added to the device's `config.json`
+file, [sshKeys section][config-json-ssh]. When the username matches a valid
+{{$names.cloud.lower}} user account, authentication is also performed against that user's
+public SSH keys [stored in {{$names.cloud.lower}}][add-ssh-key]. The username can be found in
+the profile or preferences section of the web dashboard, or with the `{{$names.company.short}}
+whoami` CLI command.
+
+Development {{ $names.os.lower }} images allow unauthenticated access and should never be
+directly exposed to the public internet.
+
+The IP address will typically be a private IP address of a local network. For remote devices,
+see [{{ $names.company.short }} tunnel][balena-tunnel].
+
 ## {{ $names.company.short }} tunnel
 
-The SSH server of the host OS will always be accessible on TCP port `22222` unless this has been blocked at the network level. To get around this, you can use the `{{ $names.company.short }} tunnel` command of the {{ $names.cli.lower }}, which may be used to map local ports to listening ports on the device. For example, the following command maps the local port `4321` to port `22222` on the device:
```

Comment:
I didn't like the previous statement _"will always be accessible on TCP port `22222`"_ because it sounds like a forever guarantee, I think unintentional. Guarantees may unnecessarily prevent our future selves from making some changes, and therefore should not be distributed lightly. :-)